crates/ruff/src/printer.rs

```diff
@@ -458,17 +389,132 @@ impl Printer {
+        &self,
+        writer: &mut dyn Write,
+        diagnostics: &Diagnostics,
+        context: &EmitterContext,
+        fixables: Option<&FixableStatistics>,
+        preview: bool,
+        write_mode: WriteMode,
+    ) -> Result<()> {
+        match self.format {
+            OutputFormat::Json => {
+                let config = DisplayDiagnosticConfig::default()
+                    .format(DiagnosticFormat::Json)
+                    .preview(preview);
+                let value = DisplayDiagnostics::new(context, &config, &diagnostics.inner);
+                write!(writer, "{value}")?;
+            }
+            OutputFormat::Rdjson => {
+                let config = DisplayDiagnosticConfig::default()
+                    .format(DiagnosticFormat::Rdjson)
+                    .preview(preview);
+                let value = DisplayDiagnostics::new(context, &config, &diagnostics.inner);
+                write!(writer, "{value}")?;
+            }
+            OutputFormat::JsonLines => {
+                let config = DisplayDiagnosticConfig::default()
+                    .format(DiagnosticFormat::JsonLines)
+                    .preview(preview);
+                let value = DisplayDiagnostics::new(context, &config, &diagnostics.inner);
+                write!(writer, "{value}")?;
+            }
+            OutputFormat::Junit => {
+                let config = DisplayDiagnosticConfig::default()
+                    .format(DiagnosticFormat::Junit)
+                    .preview(preview);
+                let value = DisplayDiagnostics::new(context, &config, &diagnostics.inner);
+                write!(writer, "{value}")?;
+            }
+            OutputFormat::Concise | OutputFormat::Full => match write_mode {
+                WriteMode::Once => {
+                    TextEmitter::default()
+                        .with_show_fix_status(show_fix_status(self.fix_mode, fixables))
+                        .with_show_fix_diff(self.flags.intersects(Flags::SHOW_FIX_DIFF))
+                        .with_show_source(self.format == OutputFormat::Full)
+                        .with_unsafe_fixes(self.unsafe_fixes)
+                        .with_preview(preview)
+                        .emit(writer, &diagnostics.inner, context)?;
+
+                    if self.flags.intersects(Flags::SHOW_FIX_SUMMARY) {
+                        if !diagnostics.fixed.is_empty() {
+                            writeln!(writer)?;
+                            print_fix_summary(writer, &diagnostics.fixed)?;
+                            writeln!(writer)?;
+                        }
+                    }
+
+                    self.write_summary_text(writer, diagnostics)?;
+                }
+                WriteMode::Continuous => {
+                    TextEmitter::default()
+                        .with_show_fix_status(show_fix_status(self.fix_mode, fixables))
+                        .with_show_source(self.format == OutputFormat::Full)
```

Comment:
This should also be


```suggestion
                        .with_show_source(preview)
```

if we want to maintain the current behavior, which I think we should do until a minor release.